RecoLocalTracker/SiPixelRecHits/interface/PixelCPEBase.h

```diff
@@ -49,6 +49,46 @@
 #include <atomic>
 #endif
 
+class PixelCPEWarningSummary {
+private:
+  bool m_debug;
+  std::string m_category;
+  std::map<std::string, std::size_t> m_warnings;
+
+public:
+  PixelCPEWarningSummary(const std::string& category, bool debug = false) : m_debug(debug), m_category(category) {}
+
+  void add(const std::string& message, const std::string& details) {
+    const auto wIt = std::find_if(
```

Comment:
Here you have taken a map, which naturally has O(log(n)) search, and written an O(n) search algorithm.
Please use: `auto wIt = m_warnings.find(message);` and then compare to `m_warnings.end()`

Reply:
Sorry, I based this code on the strip unpacker which used this method (https://github.com/cms-sw/cmssw/blob/master/EventFilter/SiStripRawToDigi/plugins/WarningSummary.cc) 

I have fixed this now and someone may want to make a PR for the strip unpack to apply a similar change

Reply:
It looks like that code originally used `vector<pair<...>>`, and the algorithm wasn't improved when the data type was changed to `map`. I agree it would be good to update that code in a separate PR.

Reply:
done at https://github.com/cms-sw/cmssw/pull/32280